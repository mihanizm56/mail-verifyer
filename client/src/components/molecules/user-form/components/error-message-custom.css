.error-message-custom {
    text-align: center;
    font-size: var(--font-size-error);
    color: var(--color-error-text)
}