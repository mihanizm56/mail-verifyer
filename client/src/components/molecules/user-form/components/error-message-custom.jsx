import React from "react";
import "./error-message-custom.css";

export const ErrorMessageCustom = ({ submissionError, error }) => {
	const errorText = Boolean(error) ? error : submissionError;

	return <span className="error-message-custom">{errorText}</span>;
};
