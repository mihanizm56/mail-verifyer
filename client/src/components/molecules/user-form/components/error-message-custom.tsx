import React from "react";
import "./error-message-custom.css";

type ErrorMessageCustomProps = {
	submissionError: string,
	error: string | false,
};

export const ErrorMessageCustom = ({ submissionError, error }: ErrorMessageCustomProps) => {
	const errorText = Boolean(error) ? error : submissionError;

	return <span className="error-message-custom">{errorText}</span>;
};
