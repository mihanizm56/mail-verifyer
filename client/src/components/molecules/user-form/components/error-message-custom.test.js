import React, { createElement } from "react";
import { shallow } from "enzyme";
import { ErrorMessageCustom } from "./error-message-custom";

describe("test ErrorMessageCustom", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				submissionError: "submission error",
				error: "error",
			};
			const component = shallow(<ErrorMessageCustom {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render error text when error prop was given", () => {
			const props = {
				submissionError: "submission error",
				error: "error",
			};
			const component = shallow(<ErrorMessageCustom {...props} />);
			const textValue = component.find(".error-message-custom").props().children;

			expect(textValue).toEqual(props.error);
		});
		it("render error text when error prop wa not given", () => {
			const props = {
				submissionError: "submission error",
			};
			const component = shallow(<ErrorMessageCustom {...props} />);
			const textValue = component.find(".error-message-custom").props().children;

			expect(textValue).toEqual(props.submissionError);
		});
	});
});
