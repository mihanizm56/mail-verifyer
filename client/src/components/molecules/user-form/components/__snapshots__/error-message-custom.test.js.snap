// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorMessageCustom test render contract render snap with all props 1`] = `
<span
  className="error-message-custom"
>
  error
</span>
`;
